src/include/device.h

```diff
@@ -530,7 +535,7 @@ inline int ncclDevFuncId(int coll, int devRedOp, int type, int algo, int proto)
 
   // <all_algos> / <all_protos> / <all_redops> / <all_types>
   if (coll == ncclFuncAllReduce) {
-    row += (((algo * NCCL_NUM_PROTOCOLS + proto) * ncclNumDevRedOps + devRedOp) * ncclNumTypes + type) - /*floats for each SumPostDiv*/ 4 * (algo * NCCL_NUM_PROTOCOLS + proto);
+    row += (((algo * NCCL_NUM_PROTOCOLS + proto) * ncclNumDevRedOps + devRedOp) * ncclNumTypes + type) - /*floats for each SumPostDiv*/ 6 * (algo * NCCL_NUM_PROTOCOLS + proto);
     goto have_row;
   }
   row += (NCCL_NUM_ALGORITHMS - 2) * NCCL_NUM_PROTOCOLS * (ncclNumDevRedOps * ncclNumTypes - /*floats for each SumPostDiv*/ 4);
```

Comment:
num floats for each SumPostDiv is now 6. need update this number. This should also be done for both Reduce and ReduceScatter too.

Reply:
Confirmed that most of UT passed after function index fix as Bertan pointed out:  cdc1820.
But now there are failures with FP8 data accuracy check in UT

Reply:
It seems the issues are with ManagedMemGraph, which are not a common use cases. We can skip it for now and debug later.

Reply:
update SumPostDiv number to 6 for both Reduce and ReduceScatter according to cr comments.